components/automate-cli/cmd/chef-automate/cleanup.go

```diff
@@ -0,0 +1,172 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"strings"
+	"github.com/spf13/cobra"
+	"golang.org/x/crypto/ssh"
+)
+
+var cleanupFlags = struct {
+	onprem        bool
+}{}
+
+type SshClient struct {
+	sshUser string
+	sshPort string
+	sshKeyFile string
+}
+
+func init() {
+	RootCmd.AddCommand(cleanupCmd)
+	cleanupCmd.PersistentFlags().BoolVar(&cleanupFlags.onprem, "onprem", false, "Cleaning up all the instances related to onprem ")
+
+}
+
+var cleanupCmd = &cobra.Command {
+		Use:  "cleanup",
+	    Short: "cleanup the Automate HA instances",
+	    Long:  "cleaning up the instance of all the Automate HA related Applications.",
+		Annotations: map[string]string{
+			NoCheckVersionAnnotation: NoCheckVersionAnnotation,
+		},
+		RunE: runCleanupCmd,
+		Hidden: true,
+		
+}
+
+const (
+		FRONTENDCLEANUP_COMMANDS = `
+		sudo systemctl stop chef-automate;
+		sudo rm -rf /hab;
+		`
+	
+		BACKENDCLEANUP_COMMANDS = `
+		sudo systemctl stop hab-sup;
+		sudo rm -rf /hab;   
+		`
+)
+
+func runCleanupCmd(cmd *cobra.Command, args []string) error {
+	infra, err := getAutomateHAInfraDetails()
+	if err != nil {
+		return err
+	}
+	if infra != nil {
+		sshInfo :=SshClient{
+			sshUser: infra.Outputs.SSHUser.Value,
+			sshPort: infra.Outputs.SSHPort.Value,
+			sshKeyFile: infra.Outputs.SSHKeyFile.Value,
+		}
+
+			writer.Printf(strings.Join(args, ""))
+		if isA2HARBFileExist() {
+			if cleanupFlags.onprem {
+				automateIps := infra.Outputs.AutomatePrivateIps.Value
+				chefserverIps := infra.Outputs.AutomatePrivateIps.Value
+				postgresqlIps := infra.Outputs.PostgresqlPrivateIps.Value
+				opensearchIps := infra.Outputs.OpensearchPrivateIps.Value
+				for i := 0; i < len(automateIps); i++ {
+					servername := "Automate"
+					err := executeCleanupOnRemote(sshInfo, automateIps[i], servername, FRONTENDCLEANUP_COMMANDS)
+					if err !=nil {
+						writer.Errorf("%s", err)
+						return err
+					}
+				}
+				for i := 0; i < len(chefserverIps); i++ {
+					servername := "chef server"
+					err := executeCleanupOnRemote(sshInfo, chefserverIps[i], servername, FRONTENDCLEANUP_COMMANDS)
+					if err !=nil {
+						writer.Errorf("%s", err)
```

Comment:
use writer.Error(err), because if error is not in string formate, then formate will crash the code

Reply:
done